mono/metadata/loader.c

```diff
@@ -2418,9 +2418,11 @@ mono_loader_unlock_if_inited (void)
  *
  * Return the signature of the method M. On failure, returns NULL, and ERR is set.
  */
-MonoMethodSignature*
+MONO_PROFILER_API MonoMethodSignature*
```

Comment:
If the declaration in the `.h` file is marked as `MONO_PROFILER_API` then there is no need to do it in the `.c` file too.

Reply:
I find this easier to read and understand -- I want MONO_API_ERROR_INIT to be very near MONO_API. However, if the .h has actual build meaching and .c does not, then it is just untrusted comments and I'm less sure. This PR isn't really ready I think. Still have to run tests and see how close I am.. (if I remove all the plain old error_init, but still, want to rerereview everything myself again).